internal/compiler-bridge/src/main/scala/xsbt/CallbackGlobal.scala

```diff
@@ -78,6 +78,16 @@ sealed class ZincCompiler(settings: Settings, dreporter: DelegatingReporter, out
     extends CallbackGlobal(settings, dreporter, output)
     with ZincGlobalCompat {
 
+  override def getSourceFile(f: AbstractFile): BatchSourceFile = {
+    val file = f match {
+      case plainFile: PlainFile =>
+        val vf = callback.asInstanceOf[AnalysisCallback3].toVirtualFile(plainFile.file.toPath)
+        new ZincPlainFile(vf.asInstanceOf[PathBasedFile])
```

Comment:
The conversion logic is... hacky.

I would appreciate any feedback suggesting a better solution.